File: Notes/5-5_LevelOrderTraversal.py
# binary tree level order traversal
# python implementation

class treeNode:
    def __init__(self, val):
        self.val = val
        self.left = None
        self.right = None

    def __repr__(self):
        return str(self.val)


def levelOrderTraversal(root):
    '''
    A function that traverse a binary tree in level order.
    '''
    queue, seque = [root], []
    curr = root
    while queue:
        curr = queue[0]
        if curr.left:
            queue.append(curr.left)
        if curr.right:
            queue.append(curr.right)
        print(curr, end=' ')
        queue.pop(0)


        

File: FDSC/Notes/5-5_LevelOrderTraversal.c
/* lever order traversal of binary tree */
#define MAXSIZE 100

typedef struct Node *TreeNode;
struct Node {
    int data;
    TreeNode left;
    TreeNode right;
};

void level_order(TreeNode ptr)
{
	int front = rear = 0;
	TreeNode queue[MAXSIZE];
	if (!ptr)
		return;
	enqueue(front, &rear, ptr);
	for (;;)
	{
		ptr = dequeue(&front, rear);
		if (ptr)
		{
			printf("%d ", ptr->data);
			if (ptr->left)
				enqueue(front, &rear, ptr->left);
			if (ptr->right)
				enqueue(front, &rear, ptr->right);
		}
		else
			break;
	}
}